### PR TITLE
Fix building with Java 11 and stop using Depstar

### LINKED ISSUE #48390: Migrate build from depstar
We should migrate from depstar since it's deprecated and, in particular, prevents updating `org.flatland/ordered`.

### PR DESCRIPTION
**Pull request by camsaul**
Removes usage of depstar, now we use `tools.build` directly.
Fixes #48390

Fixes issues with tools like `jar uf` not working with Java 11 -- makes https://github.com/metabase/metabase-ee-extra/issues/44 fixable

The main fix here is to strip out `META-INF/versions/*/module-info.class` files from the artifact since we don't have a modular jar, aren't doing anything fancy to merge the module files, and this breaks stuff like `jar uf` on older java versions. 

See https://metaboat.slack.com/archives/C5XHN8GLW/p1731631694802179?thread_ts=1731504670.951389&cid=C5XHN8GLW for more info

I was able to debug this all from the REPL (thanks Clojure build scripts! see also #50061) and am checking in some extra tools I wrote to debug this stuff -- `dev.build.check-class-file-versions` is not part of the fix but is useful for debugging so I'll leave it in.

### CODE REVIEW

File: dev/src/dev/build/check_class_file_versions.clj [hunk 1]
```diff
@@ -0,0 +1,109 @@
+(ns dev.build.check-class-file-versions
```
Comment (camsaul) [inline dev/src/dev/build/check_class_file_versions.clj:1]: Not part of the fix -- this is code I used for debugging it tho

**Comment by dpsutton** [comment]
I think some things are missing from the jar.

In the test failures are lots of errors like

```
%d [%thread] %-5level %logger - %msg%n java.io.FileNotFoundException: /metabase/email/dashboard_subscription.hbs
	at com.github.jknack.handlebars.io.URLTemplateLoader.sourceAt(URLTemplateLoader.java:70) ~[metabase.jar:?]
	at com.github.jknack.handlebars.Handlebars.compile(Handlebars.java:500) ~[metabase.jar:?]
	at com.github.jknack.handlebars.Handlebars.compile(Handlebars.java:481) ~[metabase.jar:?]
	at metabase.channel.template.handlebars$render.invokeStatic(handlebars.clj:41) ~[metabase.jar:?]
...
```

This file is in https://github.com/metabase/metabase/tree/master/src/metabase/email. It's in the jar made by the latest commit on master but it's not in the jar made here for some reason.

**Comment by camsaul** [comment]
Ok the switch from `depstar` to tools.build is probably doing something unexpected

**Comment by camsaul** [comment]
Ok I concluded that the `tools.build` version of `uber` doesn't copy resources the way the Depstar did, you have to do it as part of `b/copy-dir` (which we were already doing for `resources`)... I'll update the build script and it should be fixed

**Comment by piranha** [comment]
The test failures seems weird and feel like there is nothing to do with the changes themselves. :\

**Review by sansolovyov — APPROVED** [review]
Seems to be working now :)

**Pull request merged**